okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/OkHttpClients.java

```diff
@@ -119,7 +119,7 @@ private static RemotingOkHttpClient createInternal(
         client.sslSocketFactory(config.sslSocketFactory(), config.trustManager());
 
         // Intercept calls to augment request meta data
-        client.addNetworkInterceptor(InstrumentedInterceptor.create(registry, hostMetrics, serviceClass));
+        client.addInterceptor(InstrumentedInterceptor.create(registry, hostMetrics, serviceClass));
```

Comment:
connecting to a host occurs before the network interceptors are invoked which means some IOExceptions won't be seen by network interceptors. think moving it here is fine because we'll only record metrics for the responses the application actually sees.

Reply:
would a test fail if this were a network interceptor?

Reply:
yes. the one added to OkHttpClientsTest